src/csharp/Grpc.IntegrationTesting/ServicesGrpc.cs

```diff
@@ -47,12 +47,28 @@ public interface IBenchmarkServiceClient
     }
 
     // server-side interface
+    [System.Obsolete("Service implementations should inherit from the generated abstract base class instead.")]
     public interface IBenchmarkService
     {
       Task<global::Grpc.Testing.SimpleResponse> UnaryCall(global::Grpc.Testing.SimpleRequest request, ServerCallContext context);
       Task StreamingCall(IAsyncStreamReader<global::Grpc.Testing.SimpleRequest> requestStream, IServerStreamWriter<global::Grpc.Testing.SimpleResponse> responseStream, ServerCallContext context);
     }
 
+    // server-side abstract class
+    public abstract class BenchmarkServiceBase
```

Comment:
I didn't use just "Health" because I think for many services the class name might collide with the namespace name (e.g. service math.Math which i think a pretty common pattern in .proto files). We didn't have this problem with the interface that has the "I" prefix.
Do you see an issue with class name being the same as the package name?